packages/portfolio-contract/test/planner.exo.test.ts

```diff
@@ -118,3 +110,98 @@ test('planner exo submit method', async t => {
     vt.when(planner.resolvePlan(portfolioId, 1, plan, 1, 2)),
   );
 });
+
+test('planner can reject a plan due to insufficient funds', async t => {
+  const zone = makeHeapZone();
+
+  const vt = prepareVowTools(zone);
+  // Mock dependencies with minimal implementation
+  const mockRebalance = (_seat, offerArgs, _kit) => {
+    t.log('rebalance called with', offerArgs);
+    return vt.asVow(() => undefined);
+  };
+
+  const mockZcf = {
+    makeEmptySeatKit: () => ({
+      zcfSeat: null as any,
+    }),
+  } as ZCF;
+
+  const board = makeFakeBoard();
+  const storage = makeFakeStorageKit('published', { sequence: true });
+  const { getPortfolioStatus } = makeStorageTools(storage);
+  const marshaller = board.getReadonlyMarshaller();
+  const makePortfolio = preparePortfolioKit(zone, {
+    usdcBrand: USDC,
+    marshaller,
+    portfoliosNode: storage.rootNode
+      .makeChildNode('ymax0')
+      .makeChildNode('portfolios'),
+    vowTools: vt,
+    ...({} as any),
+  });
+  const aPortfolio = makePortfolio({ portfolioId: 1 });
+  const mockGetPortfolio = _id => aPortfolio;
+
+  // Create planner exo
+  const makePlanner = preparePlanner(zone, {
+    rebalance: mockRebalance,
+    zcf: mockZcf,
+    getPortfolio: mockGetPortfolio,
+    shapes: makeOfferArgsShapes(USDC),
+    vowTools: vt,
+  });
+
+  const planner = makePlanner();
+
+  // Set up portfolio state
+  aPortfolio.manager.setTargetAllocation({ USDN: 100n });
+
+  const portfolioId = 0;
+  const amount = { brand: USDC, value: 100n };
+
+  // Start a flow that will be waiting for a plan and simulate proper cleanup
+  const { stepsP, flowId } = aPortfolio.manager.startFlow({
+    type: 'withdraw',
+    amount,
+  });
+
+  {
+    const {
+      policyVersion,
+      rebalanceCount,
+      flowsRunning = {},
+    } = await getPortfolioStatus(1);
+    t.log('before reject:', { policyVersion, rebalanceCount, flowsRunning });
+    t.is(Object.keys(flowsRunning).length, 1, 'should have one running flow');
+    t.is(rebalanceCount, 0, 'rebalanceCount should start at 0');
+  }
+
+  // Planner rejects the plan due to insufficient funds
+  t.notThrows(() => planner.rejectPlan(portfolioId, 1, 'insufficient funds'));
+
+  // Verify the flow's promise gets rejected with the expected error
+  await t.throwsAsync(() => vt.when(stepsP), { message: 'insufficient funds' });
```

Comment:
Nit: [`throwsAsync`](https://github.com/avajs/ava/blob/main/docs/03-assertions.md#throwsasyncthrower-expectation-message) accepts a promise, so this IIFE is unnecessary.
```suggestion
  // Verify the flow's promise gets rejected with the expected error
  await t.throwsAsync(vt.when(stepsP), { message: 'insufficient funds' });
```